hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -2434,13 +2434,18 @@
     {
       name: chip_sw_keymgr_sideload_kmac_error
       desc: '''
-            Verify the effect of KMAC returning an error during keymgr sideload operation.
+            Verify the effect of KMAC returning an error during a keymgr operation.
 
-            - Follow the steps in `chip_sw_keymgr_sideload_kmac` test.
+            - Configure keymgr to enter any of the 3 working states.
+            - Issue a keymgr operations.
             - While the KMAC is actively computing the digest, glitch the KMAC app sparse FSM to
               trigger a fault.
-            - Verify that KMAC returns an error signal to the keymgr.
-            - TBD
+            - Verify that KMAC returns an error signal to the keymgr via checking keymgr CSRs, when
+              the operation is done:
+              - Check `op_status` is set to `DONE_ERROR`.
+              - Check `fault_status.kmac_done` is set to 1.
```

Comment:
This might be out of the scope of the test plan, but...

I couldn't figure out how faulting `kmac_app`'s FSM propagates to keymgr's error/fault CSR. I wonder which of the following is wrong about my reasoning:

1. Keymgr starts KMAC operation, and `kmac_if` moves to `StOpWait` after sending all message chunks. Note that `kmac_data_i.error` here will propagate to `keymgr_ctrl` when `kmac_data_i.done` is raised to 1. So, this is condition on `kmac_app` raising `done` signal.
2. On the KMAC side, faulting `kmac_app`'s FSM causes `kmac_data_i.error = 1`, but `kmac_data_i.done` is not raised to 1 (because faulty state is `StTerminalError`).
3. So glitching `kmac_app`'s FSM leads to `kmac_if` being stuck at `StOpWait`, and `kmac_app` sending `error`, but not `done`.

I might be missing something here because I couldn't properly go through `keymgr_ctrl` yet.

Reply:
@eunchan yeah i think your observation is right. 
In the case of a glitched fsm, I guess the kmac interface would just never complete then? Maybe we need to change this to glitch an error, or update the kmac app fsm behavior to always say "done and error" when in error state.

Reply:
yeah i had a quick look.  It looks like we either need to make a `keymgr` update to always look at error regardless of `done`, or update `kmac` to output `done` whenever it's in the same error state. 

The latter feels a bit more correct to me, but let me know what you think.
@cindychip

Reply:
The latter sounds good to me.

Reply:
I think the first approach is better. Consider this:

1. `kmac_app` FSM glitch happens before all message chunks are sent, so assume that `kmac_if` FSM is at `StTx` when glitch happens.
2. If we follow the second approach, I think `kmac_if` will be stuck at `StTx` even if both `error` and `done` are 1.

I think `kmac_if` FSM should always move to error state if `error` is asserted by `kmac_app`, regardless of `done` signal.

Reply:
o i assumed that kmac_app even in error state will just allow transactions to flush through so things "look" as if they are normal.

Reply:
I agree that setting both done and error in error state makes the operation look normal.
@eunchan WDYT?

Reply:
I agree with Tim and Weicai. 

If fault attacks cause the error, then I expect KMAC fires alerts, which eventually leads to SoC reset. Will this change (kmac asserting done and error) matter?

Reply:
> If fault attacks cause the error, then I expect KMAC fires alerts, which eventually leads to SoC reset. Will this change (kmac asserting done and error) matter?

I guess it doesn't matter much. Setting `done` makes the keymgr look normal and may be slightly easier to test.
If we prefer to keep it as is, we need to think of another way to verify that the `error` signal is received and processed correctly by keymgr in a chip-level test.
Any thoughts? @tjaychen @ballifatih

Reply:
I didn't mean I want to keep as it is. Changing to assert done is good idea. But as Fatih mentioned, when error occurs, we may design the FSM cautiously to accept the incoming transaction and asserting the error and done in pair with the request. not blindly asserting the done and error altogether.

Reply:
ah i thought @ballifatih was suggesting making the "error" signal more like an interrupt into `keymgr`.
I think that would be fine too.  I suggested the `done` approach mostly because it keeps the interface assumptions the same and would allow us to get around cases of being more flexible when the error condition when the transaction isn't done (like a bus).  

@weicaiyang can chime in, but if we make the error more like an interrupt, we would also need to add a test case probably on keymgr block level, and it would be nice to avoid that if we can.

Reply:
Agree with @tjaychen, if the error becomes an interrupt, DV needs to verify it happens in any place (not just along with done). I slightly prefer to either send a `done` with `error`, or keep it as is so that keymgr gets stuck and the fatal alert eventually leads to a reset.

Reply:
yeah, we need to have the `ready` too. And the early assertion of done will also lead to done consistency error too.
https://cs.opensource.google/opentitan/opentitan/+/master:hw/ip/keymgr/rtl/keymgr_kmac_if.sv;l=336?q=KMAC_IF_DONE.CTRL.CONSISTENCY&ss=opentitan%2Fopentitan:hw%2Fip%2Fkeymgr%2Frtl%2F